### PR TITLE
(feat): JSON/YAML support for input data

### PR DESCRIPTION
**Pull request by hopefulTex**
- Input data now exported in a struct
- Missing JSON fields populated with defaults
- JSON fields can be overridden by flags
TODO:
[]Human-readable JSON parsing errors

### CODE REVIEW

**Comment by maaslalani** [comment]
This is super awesome! Thank you so much @hopefulTex, do you think it's reasonable to support YAML as well?

**Comment by hopefulTex** [comment]
definitely, YAML looks like a simple drop-in now that it's structured for JSON.

**Comment by hopefulTex** [comment]
Either I'm missing something big, or the yaml package devs are beautiful amazing simplistic people.
Please let me know if it's the former.

**Comment by TerraSkye** [comment]
@hopefulTex  you need to ad `yaml` tag in addtion to the `json` tag to make it work

**Comment by hopefulTex** [comment]
> @hopefulTex you need to ad `yaml` tag in addtion to the `json` tag to make it work

I think the yaml library can use/accept the json tags, but technically correct is the best correct.

**Comment by hopefulTex** [comment]
It's not very clean, but it works once more.
Just a matter of saving flags before unmarshalling then emplacing them correctly once more.
I can probably come up with a cleaner solution with some research in the json library code.

File: import.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+)
+
+func importData(path string, structure *invoiceData) error {
+	fileText, err := os.ReadFile(path)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	length := len(path)
+	if path[length-5:] == ".json" {
```
[the remark this answers is not in the document]
Comment (hopefulTex) [reply import.go:31]: Would you believe me if I told you I thought about it hard enough that I imagined that I did.

[the remark this answers is not in the document]
Comment (maaslalani) [reply import.go]: Hahaha totally, happens to me to!

**Pull request merged**